plugins/TrollManagement/class.trollmanagement.plugin.php

```diff
@@ -469,4 +499,189 @@ private function checkTroll(int $userID, array &$args) {
+            'TrollManagement.PerFingerPrint.Enabled' => c('TrollManagement.PerFingerPrint.Enabled', false),
+            'TrollManagement.PerFingerPrint.MaxUserAccounts' => c('TrollManagement.PerFingerPrint.MaxUserAccounts', 5),
+        ]);
+
+        $sender->Form->setModel($configurationModel);
+
+        // If seeing the form for the first time...
+        if ($sender->Form->authenticatedPostBack() === false) {
+            $sender->Form->setData($configurationModel->Data);
+        } else {
+            $configurationModel->Validation->applyRule('TrollManagement.PerFingerPrint.Enabled', 'Boolean');
+
+            if ($sender->Form->getFormValue('TrollManagement.PerFingerPrint.Enabled')) {
+                $configurationModel->Validation->applyRule('TrollManagement.PerFingerPrint.MaxUserAccounts', 'Required');
+                $configurationModel->Validation->applyRule('TrollManagement.PerFingerPrint.MaxUserAccounts', 'Integer');
+                $configurationModel->Validation->applyRule(
+                    'TrollManagement.PerFingerPrint.MaxUserAccounts',
+                    'validatePositiveNumber',
+                    sprintf(t('%s must be a positive number.'), t("Maximum user's accounts"))
+                );
+            }
+
+            if ($sender->Form->save()) {
+                $sender->StatusMessage = t('Your changes have been saved.');
+            }
+        }
+
+        $sender->render($this->getView('configuration.php'));
+    }
+
+    /**
+     * Add a _probable_ justification as to why a user's is on the applicant's list if there are too many user
+     * accounts using the same fingerprint.
+     *
+     * @param Controller $sender
+     * @param array $args
+     */
+    public function base_applicantInfo_handler($sender, $args) {
+        if (c('TrollManagement.PerFingerPrint.Enabled', false)) {
+            $maxSiblingAccounts = c('TrollManagement.PerFingerPrint.MaxUserAccounts');
+            $userFingerprint = val('Fingerprint', $args['User']);
+            if (!empty($userFingerprint)) {
+                $fingerprintUsages = $this->getSharedFingerprintsUsersCount($userFingerprint);
+                if ($fingerprintUsages >= $maxSiblingAccounts) {
+                    $sender->EventArguments['ApplicantMeta'][t("Fingerprint issue")] = sprintf(
+                        t("%s accounts are sharing the '%s' fingerprint."),
+                        $fingerprintUsages,
+                        $userFingerprint
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Return a count of users using the same provided fingerprint.
+     *
+     * @param null|string $fingerprint
+     * @return int
+     */
+    public function getSharedFingerprintsUsersCount($fingerprint): int {
+        if (!is_null($fingerprint)) {
+            $sql = clone Gdn::sql();
+            $sql->reset();
+            $users = $sql
+                ->select('userID AS siblingsCount', 'count')
```

Comment:
As often as this query has the potential to be run, trying to perform a count each time isn't going to be so performant. It looks like you're only ever comparing this against the "max siblings". Maybe just query the user table for rows with that fingerprint, limiting by the "max siblings" value. If you have at least that many, we know this fingerprint has exceeded its account allowance.

Reply:
Alright. This has been adjusted.